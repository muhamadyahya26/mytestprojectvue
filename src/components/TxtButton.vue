<template>
  <div>
    <label class="lbl-txt">{{ label }}</label>
    <button class="btn-action" @click.prevent="changeText">Contoh tombol</button>
  </div>
</template>

<script>
export default {
  name: 'TxtButton',
  data() {
    return {
      label: 'Hai',
    }
  },
  methods: {
    changeText() {
      this.label = 'Hello';
    }
  }
}
</script>
