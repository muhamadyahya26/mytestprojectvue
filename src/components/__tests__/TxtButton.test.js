import { shallowMount } from '@vue/test-utils';
import TxtButton from '../TxtButton.vue';

describe('TxtButton.vue', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallowMount(TxtButton);
  });

  afterEach(() => {
    wrapper.destroy();
  })

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  })

  it('should change text label when button clicked', () => {
    wrapper.find('.btn-action').trigger('click.prevent');
    
    expect(wrapper.vm.label).toBe('Hello'); 
  })  
});
