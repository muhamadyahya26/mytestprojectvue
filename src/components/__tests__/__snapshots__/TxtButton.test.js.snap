// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxtButton.vue should render correctly 1`] = `
<div>
  <label
    class="lbl-txt"
  >
    Hai
  </label>
   
  <button
    class="btn-action"
  >
    Contoh tombol
  </button>
</div>
`;
